libs/ledger-live-common/src/families/avalanchepchain/specs.ts

```diff
@@ -0,0 +1,93 @@
+import invariant from "invariant";
+import type {
+  AvalanchePChainAccount,
+  AvalanchePChainResources,
+  Transaction,
+} from "./types";
+import { getCryptoCurrencyById, parseCurrencyUnit } from "../../currencies";
+import type { AppSpec } from "../../bot/types";
+import { DeviceModelId } from "@ledgerhq/devices";
+import { canDelegate } from "./utils";
+import expect from "expect";
+import { BigNumber } from "bignumber.js";
+import {
+  FIGMENT_AVALANCHE_VALIDATOR_NODES,
+  TWO_WEEKS,
+  FIVE_MINUTES,
+  MINUTE,
+} from "./utils";
+import { acceptTransaction } from "./speculos-deviceActions";
+
+const currency = getCryptoCurrencyById("avalanchepchain");
+const minimalAmount = parseCurrencyUnit(currency.units[0], "25");
+const validator = FIGMENT_AVALANCHE_VALIDATOR_NODES[0];
+
+const avalanche: AppSpec<Transaction> = {
+  name: "Avalanche",
+  currency: getCryptoCurrencyById("avalanchepchain"),
+  appQuery: {
+    model: DeviceModelId.nanoS,
+    appName: "Avalanche",
+  },
+  genericDeviceAction: acceptTransaction,
+  testTimeout: 2 * 60 * 1000,
+  mutations: [
```

Comment:
is it possible to do a "Send" on this integration? I would expect coin integration first implements send flow and as I don't see it mentioned in the bot specs, it means you would have to manually send the funds to create a few accounts on the bot seed?

if there is a send half funds defined, the bot would be able to automatically rotate funds between accounts.

also it is important for the bot to be able to test things over time, what happens to the delegated funds, are they locked? is it possible to undelegate/redelegate? because in the present case, i wonder if that mutation will be only be able to run once and that's it?

Reply:
Hey @gre , good questions.

It isn't possible to do a "send" transaction in this integration; users cannot send from one P chain address to another. In order to get funds on the P chain, a user needs to perform a "cross-chain transfer" from either the X chain or the C chain. The cross-chain transfer feature is outside the scope of this pull request.

It also isn't possible to undelegate or redelegate on the Avalanche blockchain, and funds are locked for a minimum of two weeks. Avalanche also requires a minimum of 25 AVAX tokens per staking transaction, so the bot must have at least 25 tokens available in order for this test to pass.

Reply:
then i'm not sure that this bot would be relevant to exists in the current state because that means the bot will have nothing to do if it can only delegate once 🤔  cc @haammar-ledger

Reply:
Hi, thanks for insights from both of you.
@trentkrogers acknowledged that this integration won't be able to send, only receive. And if I understood correctly, that there is no action available to the user after they delegated.
Follow-up question: what happens to the staked AVAX after the 2 weeks? No user action required?

Reply:
@haammar-ledger That's right, after delegation no user action is available or required. The delegation will automatically end at the time the user specified during the initial staking process.

Reply:
@trentkrogers So to conclude on this, all the bot could do is a `delegate` transaction every 2 weeks for each account, correct?
So in theory we could have a dozen accounts rotating so that almost everyday one performs a delegate.

I think this would still make sense over no bot at all, however two main issues are:
- the amount of funds that will need, if we want let's say 10 accounts x 25 AVAX each =~ 3000€... but we can probably accommodate with less accounts.
- providing funds to the accounts is not as straightforward as usual, here each account will have to be manually funded through cross-chain transfers. But can probably be done as well.

I suggest keeping this spec as it is, it should just be ignored if there aren't appropriate accounts to use, which I think is the case with the proper `invariant` you have defined.

Reply:
@haammar-ledger That's right; this bot will require 25 AVAX to run, and will have to be funded via cross-chain transfers. It's a manual process, but I've done it before :)

I'll keep the spec as it is for now.